experimental/DoubleAndHyperComplexes/src/Morphisms/cartan_eilenberg_resolutions.jl

```diff
@@ -0,0 +1,59 @@
+mutable struct MapLifter{MorphismType} <: HyperComplexMorphismFactory{MorphismType}
+  dom::AbsHyperComplex
+  cod::AbsHyperComplex
+  orig_map::Map
+  start_index::Int
+  offset::Int
+  check::Bool
+
+  function MapLifter(::Type{MorphismType}, dom::AbsHyperComplex, cod::AbsHyperComplex, phi::Map; 
+      start_index::Int=0, offset::Int=0, check::Bool=true
+    ) where {MorphismType}
+    @assert dim(dom) == 1 && dim(cod) == 1 "lifting of maps is only implemented in dimension one"
+    @assert (is_chain_complex(dom) && is_chain_complex(cod)) || (is_cochain_complex(dom) && is_cochain_complex(cod))
+    @assert domain(phi) === dom[start_index]
+    @assert codomain(phi) === cod[start_index + offset]
+    return new{MorphismType}(dom, cod, phi, start_index, offset)
+  end
+end
+
+is_chain_complex(c::AbsHyperComplex) = (dim(c) == 1 ? direction(c, 1) == (:chain) : error("complex is not one-dimensional"))
+is_cochain_complex(c::AbsHyperComplex) = (dim(c) == 1 ? direction(c, 1) == (:cochain) : error("complex is not one-dimensional"))
+
+
+function (l::MapLifter{T})(Phi::AbsHyperComplexMorphism, I::Tuple) where {T<:ModuleFPHom}
+  i = first(I)
+  i == l.start_index && return l.orig_map
+
+  inc = 1 
+  if is_chain_complex(domain(Phi))
+    inc = -1
+  end
+  psi = Phi[Tuple(i+inc)]
+  dom_map = map(domain(Phi), 1, I)
```

Comment:
I took this from the `ComplexOfMorphisms`. But I agree! Actually, this has already lead to quite confusing error messages, because sometimes it would try to use a method of the original `map` on my input.

@fieker : Your call for the `ComplexOfMorphisms`. I will then follow your lead.